apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -48,14 +52,51 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
     const [searchParams, setSearchParams] = useSearchParams();
     const [query, setQuery] = useState('');
 
-    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
-
-    const [selectedModule, setSelectedModule] = useState(
-        !!paramModule && modulenames.includes(paramModule)
-            ? paramModule
-            : modulenames[0]
+    const getModule = (
+        searchParams: URLSearchParams,
+        setSearchParams: (
+            params: URLSearchParamsInit,
+            navigateOpts?: NavigateOptions
+        ) => void,
+        modulenames: string[]
+    ): string => {
+        const paramModule = searchParams.get('module') || null;
+
+        // A module has been specified and the package has this module
+        if (!!paramModule && modulenames.includes(paramModule)) {
+            return paramModule;
+        }
+
+        // A module has been specified and the package does not have this module
+        if (!!paramModule) {
+            const newSearchParams = new URLSearchParams(searchParams);
+            newSearchParams.delete('module');
+            setSearchParams(newSearchParams, { replace: true });
```

Comment:
Because you call `getModule` during render, this means you actually set state during render. This isn't good, as components should be pure with all side-effects being isolated into `useEffect` (or some equivalent). 

Concretely, I think you can solve this like this:

```typescript
// Get the selected module, or default to the first module
const selectedModule = modulenames.includes(searchParams.get('module')) ? searchParams.get('module') : modulenames[0];

// In an effect, manage the case where the `module` is not a valid value:
useEffect(() => {
  if (searchParams.get('module') && !modulenames.includes(searchParams.get('module'))) {
    const newSearchParams = new URLSearchParams(searchParams);
    newSearchParams.delete('module');
    setSearchParams(newSearchParams, { replace: true });
  }
}, [searchParams, setSearchParams, modulenames]);
```

This should have the correct behavior, and I think is a lot easier to reason about as well.